docs/api/supervisor/endpoints.md

```diff
@@ -2351,6 +2351,22 @@ Move datadisk to a new location, **This will also reboot the device!**
 
 </ApiEndpoint>
 
+<ApiEndpoint path="/os/datadisk/wipe" method="post">
+
+Wipe the datadisk and all user data, **This will also reboot the device!** This API requires an admin token
+
+This API will wipe all config for addons and Home Assistant and any locally
+stored data in config, backups, media, etc. The machine will reboot during this.
+
+After the reboot completes the latest stable version of Home Assistant and Supervisor
+will be downloaded. Once the process is complete you will see onboarding, like
+during initial setup.
+
+The Operating System will not change during this. This includes OS level settings
+such as network settings.
```

Comment:
As noted by @sairon, the feature actually reformats the overlay partition, hence network settings will get reset as well. So:

```suggestion
```

Reply:
Well I think we still should note that the version of the OS will not change. Which still separates this from a true factory reset. I'll remove the network settings part though.

Reply:
And maybe we should also mention that any modifications in the `boot` partition will be preserved.

Reply:
Is that something we support? I don't want to get too technical for people, users that are exclusively using the UI and CLI won't know which things they are doing cause changes in `boot` and which don't. Is there something users can change in either the HA UI or the CLI which will cause changes in `boot` we need to inform them about?

Reply:
I'm not sure what can be considered as "_what we support_" here, but it's something we sometimes have to advise people to do to resolve some issues. I'd say it's most common on RPi, where it's needed to adjust `config.txt` e.g. to make some shields work - and that's even documented in the user documentation. In the end this is the developers' docs, so I think going technical is not a bad thing :)

Reply:
Probably doesn't hurt to mention that the OS version and boot config will be preserved, but we don't need to go too much into details. I've made a suggestion.